native/cocos/engine/Engine.cpp

```diff
@@ -154,29 +157,18 @@ int Engine::restart() {
 }
 
 void Engine::close() { // NOLINT
-    if (cc::EventDispatcher::initialized()) {
-        cc::EventDispatcher::dispatchCloseEvent();
-    }
```

Comment:
Why not dispatch close event?

Reply:
dispatchCloseEvent is called before somewhere.

Reply:
@qiuguohua please double check it.

Reply:
> dispatchCloseEvent is called before somewhere.

inside onclose? The timing of these two calls is different. 'close' is an active close, and exit will be called inside, but onclose will not be called?

'onclose' is called when the window is closed

Reply:
It is called in Engine::dispatchWindowEvent, which is before Engine::close